service_container/factories.rst

```diff
@@ -293,5 +293,138 @@ previous examples takes the ``templating`` service as an argument:
                 ->args([ref('templating')])
             ;
         };
+        
+Usage example
+-------------
+
+The following example is intended to show how to create and use a factory method in Symfony framework.
+Suppose you want to realize the factory method pattern for services, that describe two delivery methods - DHL and UPS.
+
+Services (subclasses) definition::
+
+    // src/Deliveries/DHL.php
+    namespace App\Deliveries;
+
+    class DHL
+    {
+        public $costLabel;
+    
+        public function cost() {
+            return 100;
+        }
+    }
+    
+    // src/Deliveries/UPS.php
+    namespace App\Deliveries;
+
+    class UPS
+    {
+        public $costLabel;
+    
+        public function cost() {
+            return 200;
+        }
+    }
+
+Factory definition::
+
+    // src/Factories/DeliveryFactory.php
+    namespace App\Factories;
+    
+    abstract class DeliveryFactory
+    {
+        public static function create($deliveryMethod)
```

Comment:
The interface should be used here as typehint

Reply:
In case of my example I pass _the string_, which is name of subclass, but not the subclass object itself, 'cause factory creates the object based on passed string.
That's why I added typecheck inside `create()` instead:
 ```
$delivery = new $deliveryMethod;
if ( ! $delivery instanceof DeliveryInterface) {
    throw new RuntimeException(sprintf('%1$s should implement %2$s.', $deliveryMethod, DeliveryInterface::class));
}
```